node_modules/gh-users/tests/test-users.js

```diff
@@ -39,6 +41,173 @@ describe('Users', function() {
+            var email = TestsUtil.generateTestEmailAddress();
+            // Missing displayName
+            UsersTestsUtil.assertCreateUserFails(global.tests.admins.cam2014.client, null, email, 'password', {'emailPreference': 'immediate'}, 400, function() {
+
+                // Missing email
+                UsersTestsUtil.assertCreateUserFails(global.tests.admins.cam2014.client, 'displayName', null, 'password', {'emailPreference': 'immediate'}, 400, function() {
+
+                    // Invalid email
+                    UsersTestsUtil.assertCreateUserFails(global.tests.admins.cam2014.client, 'displayName', 'not an email address', 'password', {'emailPreference': 'immediate'}, 400, function() {
+
+                        // Missing password
+                        UsersTestsUtil.assertCreateUserFails(global.tests.admins.cam2014.client, 'displayName', email, null, {'emailPreference': 'immediate'}, 400, function() {
+
+                            // Invalid password
+                            UsersTestsUtil.assertCreateUserFails(global.tests.admins.cam2014.client, 'displayName', email, '12', {'emailPreference': 'immediate'}, 400, function() {
+
+                                // Invalid email preference
+                                UsersTestsUtil.assertCreateUserFails(global.tests.admins.cam2014.client, 'displayName', email, 'password', {'emailPreference': 'uwotm8'}, 400, function() {
+
+                                    // Invalid application id
+                                    TestsUtil.getGlobalAdminRestClient(function(globalAdminClient) {
+                                        UsersTestsUtil.assertCreateUserFails(globalAdminClient, 'displayName', email, 'password', {'emailPreference': 'immediate', 'app': 'Not a number'}, 400, function() {
+                                            UsersTestsUtil.assertCreateUserFails(globalAdminClient, 'displayName', email, 'password', {'emailPreference': 'immediate', 'app': -1}, 404, function() {
+                                                UsersTestsUtil.assertCreateUserFails(globalAdminClient, 'displayName', email, 'password', {'emailPreference': 'immediate', 'app': 1231231}, 404, function() {
+                                                    return callback();
+                                                });
+                                            });
+                                        });
+                                    });
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+
+        /**
+         * Verify authorization when creating a user
+         */
+        it('verify authorization when creating a user', function(callback) {
+            // Generate test tenant
+            TestsUtil.generateTestTenant(1, function(tenant, app) {
+
+                // Enable local account creation
```

Comment:
Should we also validate that an admin from tenant A cannot create a user on tenant B ?